test/e2e/instrumentation/events.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instrumentation
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	v1beta1 "k8s.io/api/events/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	eventsv1beta1 "k8s.io/client-go/kubernetes/typed/events/v1beta1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/instrumentation/common"
+
+	"github.com/onsi/ginkgo"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	eventRetryPeriod  = 1 * time.Second
+	eventRetryTimeout = 1 * time.Minute
+)
+
+func testEvent(name, label string) *v1beta1.Event {
+	someTime := metav1.MicroTime{Time: time.Unix(1505828956, 0)}
+	return &v1beta1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			Labels: map[string]string{
+				label: "true",
+			},
+		},
+		EventTime:           someTime,
+		Note:                "This is " + name,
+		Reason:              "Test",
+		Type:                "Normal",
+		ReportingController: "test-controller",
+		ReportingInstance:   "test-node",
+	}
+}
+
+func listEvents(name, namespace string, client eventsv1beta1.EventInterface) bool {
```

Comment:
listEvents name is misleading

How about:
```
func eventExistsInList(client eventsv1beta1.EventInterface, namespace, name string) bool {
  ...
}
```

Reply:
Done.